tests/framework/validators/ExistValidatorTest.php

```
@@ -295,6 +295,68 @@ public function testSecondTargetAttributeWithError(): void
+        $val = new ExistValidator(['targetRelation' => 'references', 'filter' => ['a_field' => 'non-existing']]);
+        $val->validateAttribute($m, 'id');
+        $this->assertTrue($m->hasErrors());
+    }
+
+    public function testPrepareConditionsWithAllowArrayAndArrayTargetAttribute(): void
+    {
+        $val = new ExistValidator(['allowArray' => true, 'targetAttribute' => ['id', 'name']]);
+        $m = new ValidatorTestMainModel();
+        $this->expectException('yii\base\InvalidConfigException');
+        $this->expectExceptionMessage('The "targetAttribute" property must be configured as a string.');
+        $method = new \ReflectionMethod($val, 'prepareConditions');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '323', column '23'). (undefined)

(MissingImport)